File: src/api/fastapi_app/models/log.py
from pydantic import BaseModel
from typing import List
from datetime import datetime

class LogItem(BaseModel):
    id:int
    timestamp: datetime
    endpoint: str
    method: str
    query_params: str
    request_body: str
    response_status: int
    client_ip: str

class Log(BaseModel):
    items: List[LogItem]
    total: int

File: src/frontend/app/types/log.ts
export interface LogEntry {
  id: number;
  timestamp: string;
  endpoint: string;
  method: string;
  query_params: string;
  request_body: string;
  response_status: number;
  client_ip: string;
}

export interface LogsResponse {
  items: LogEntry[];
  total: number;
}

export interface LogsPaginationParams {
  lines_per_page: number;
  page_no: number;
}
